libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select-trigger.ts

```diff
@@ -0,0 +1,758 @@
+
+  /**
+   * Strategy that is used to position the panel.
+   */
+  private _positionStrategy: DtFlexibleConnectedPositionStrategy;
+
+  /** Stream of keyboard events that can close the panel. */
+  private readonly _closeKeyEventStream = new Subject<void>();
+
+  /** Subscription to viewport size changes. */
+  // private _viewportSubscription = EMPTY.subscribe();
+
+  /** The subscription for closing actions (some are bound to document). */
+  private _closingActionsSubscription = EMPTY.subscribe();
+
+  /** The subscription for the window blur event */
+  private _windowBlurSubscription = EMPTY.subscribe();
+
+  /** Old value of the native input. Used to work around issues with the `input` event on IE. */
+  private _previousValue: string | number | null;
+
+  private _destroy$ = new Subject<void>();
+
+  constructor(
+    private _elementRef: ElementRef,
+    private _overlay: Overlay,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _viewportResizer: DtViewportResizer,
+    private _zone: NgZone,
+    private _viewportRuler: ViewportRuler,
+    private _platform: Platform,
+    private _overlayContainer: OverlayContainer,
+    @Optional() @Host() private _formField?: DtFormField<string>,
+    // tslint:disable-next-line:no-any
+    @Optional() @Inject(DOCUMENT) private _document?: any,
+    @Optional()
+    @Inject(DT_UI_TEST_CONFIG)
+    private _config?: DtUiTestConfiguration,
+    @Optional()
+    @Inject(DT_OPTION_CONFIG)
+    optionConfig?: DtOptionConfiguration,
+  ) {
+    // tslint:disable-next-line:strict-type-predicates
+    if (typeof window !== 'undefined') {
+      _zone.runOutsideAngular(() => {
+        this._windowBlurSubscription = fromEvent(window, 'blur')
+          .pipe(takeUntil(this._destroy$))
+          .subscribe(() => {
+            // If the user blurred the window while the autocomplete is focused, it means that it'll be
+            // refocused when they come back. In this case we want to skip the first focus event, if the
+            // pane was closed, in order to avoid reopening it unintentionally.
+            this._canOpenOnNextFocus =
+              document.activeElement !== this._elementRef.nativeElement ||
+              this.panelOpen;
+          });
+      });
+    }
+
+    if (this._viewportResizer) {
+      this._viewportResizer
+        .change()
+        .pipe(takeUntil(this._destroy$))
+        .subscribe(() => {
+          if (this.panelOpen && this._overlayRef) {
+            this._overlayRef.updateSize({ maxWidth: this._getPanelWidth() });
+          }
+        });
+    }
+
+    const heightConfig = calculateOptionHeight(optionConfig?.height ?? 0);
+
+    this._optionHeight = heightConfig.height;
+    this._maxPanelHeight = heightConfig.maxPanelHeight;
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+    this._closingActionsSubscription.unsubscribe();
+    this._windowBlurSubscription.unsubscribe();
```

Comment:
You already used `.takeUntil(this._destroy$)` for the `_windowBlurSubscription`. So unsubscribing is not necessary.